### PR TITLE
feat: add rfc-7

### PR DESCRIPTION
**Pull request by kuruk-mm**

### CODE REVIEW

File: rpc-csharp/src/Runtime/MessageDispatcher.cs [hunk 1]
```diff
@@ -7,15 +7,34 @@
...
+        public object message;
+        public uint messageNumber;
+
+        public ParsedMessage(RpcMessageTypes messageType, object message, uint messageNumber)
+        {
+            this.messageType = messageType;
+            this.message = message;
+            this.messageNumber = messageNumber;
+        }
+    }
```
Comment (pbosio) [inline rpc-csharp/src/Runtime/MessageDispatcher.cs]:
```suggestion
    public readonly struct ParsedMessage
    {
        public readonly RpcMessageTypes messageType;
        public readonly object message;
        public readonly uint messageNumber;

        public ParsedMessage(RpcMessageTypes messageType, object message, uint messageNumber)
        {
            this.messageType = messageType;
            this.message = message;
            this.messageNumber = messageNumber;
        }
    }
```

Comment (pbosio) [reply rpc-csharp/src/Runtime/MessageDispatcher.cs]: I think that that `struct` could be defined as `readonly` so we can be sure that C# is not creating any defensive copy

File: rpc-csharp/src/Runtime/MessageDispatcher.cs [hunk 2]
```diff
@@ -7,15 +7,34 @@
...
+    {
+        public static int ID = 0;
+        public int _id = 0;
+        public event Action<ParsedMessage> OnParsedMessage;
+
         private readonly Dictionary<string, (Action<StreamMessage>, Action<Exception>)> oneTimeCallbacks =
             new Dictionary<string, (Action<StreamMessage>, Action<Exception>)>();
 
-        private readonly ITransport transport;
+        public readonly ITransport transport;
```
Comment (pbosio) [inline rpc-csharp/src/Runtime/MessageDispatcher.cs:33]: do we need this as public?

**Review by pbosio — APPROVED** [review]

File: rpc-csharp/src/Runtime/MessageDispatcher.cs (hunk 2 above)
Comment (kuruk-mm) [reply rpc-csharp/src/Runtime/MessageDispatcher.cs:33]: > do we need this as public?

Yes. Sometimes we need to access the transport directly using the MessageDispatcher.

**Pull request merged**